spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -140,7 +140,7 @@ private void registerProperties(ConfigDataLocationResolverContext resolverContex
 
 		ConfigurableBootstrapContext bootstrapContext = resolverContext.getBootstrapContext();
 		registerSingle(bootstrapContext, KubernetesClientProperties.class, clientProperties,
-				"configDataKubernetesClientProperties");
+				"kubernetesClientProperties");
```

Comment:
this change is very interesting and I am still not very sure what is going on. If I do not rename it (so that it matches the argument of `Fabric8AutoConfiguration::kubernetesClientConfig`), when running some tests, I would get:

```
No qualifying bean of type 'org.springframework.cloud.kubernetes.commons.KubernetesClientProperties' available: expected single matching bean but found 2: spring.cloud.kubernetes.client-org.springframework.cloud.kubernetes.commons.KubernetesClientProperties,configDataKubernetesClientProperties
```

that is literally the message I get. On one hand, it makes total sense to report that `configDataKubernetesClientProperties` is already present - we register it. 

But what is `spring.cloud.kubernetes.client-org.springframework.cloud.kubernetes.commons.KubernetesClientProperties` is beyond me. 

If I create a constructor in `KubernetesClientProperties` and debug the code (so that I could create a breakpoint inside the constructor to see who is calling it and thus who is creating it), it seems that the test is trying to create another instance of this class, even if one is already registered. 

---

If I rename this field to match the name of the argument, then it will find the existing bean. I did not deal with it too much further (may be in future).